### PR TITLE
Issue 548 - Convert examples to typescript

### LINKED ISSUE #548: convert the examples to TypeScript
## Is your feature request related to a problem? Please describe

We should follow our own besst  practices.

### PR DESCRIPTION
**Pull request by gapesasi**
# The Pull Request is ready

- [x] fixes #548 
- [ ] all actions are passing
- [x] only fixes a single issue

## Overview

Converted both jsonplaceholder and reqres examples to TypeScript, making the necessary changes for it to work properly.

## Examples

- [x] the example works without any additional configuration
- [ ] the example's target has been informed and has agreed

### CODE REVIEW

**Comment by gapesasi** [comment]
@Idrinth I did not understand the check about the "examples target", how can I "inform it"?

File: examples/jsonplaceholder.typicode.com/package.json [hunk 1]
```diff
@@ -3,14 +3,19 @@
...
-    "start": "run-benchmark 2 3"
+    "start": "tsc -p tsconfig.json && run-benchmark 2 3"
   },
   "engines": {
     "node": ">=20"
   },
-  "engineStrict": true
+  "engineStrict": true,
+  "devDependencies": {
+    "ts-node": "^10.9.2",
```
Comment (Idrinth) [inline examples/jsonplaceholder.typicode.com/package.json:18]: tsc may be easier :)

File: examples/jsonplaceholder.typicode.com/src/routes/main/create-post.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-module.exports = (apiRootUrl,) => ({
+export default (apiRootUrl: string,) => ({
```
[the remark this answers is not in the document]
Comment (Idrinth) [reply examples/jsonplaceholder.typicode.com/src/routes/main/create-post.ts:1]: @coderabbitai the extra comma is a requirement in this organisation.

File: examples/reqres.in/package.json [hunk 3]
```diff
@@ -3,14 +3,19 @@
...
-    "start": "run-benchmark 2 3"
+    "start": "tsc -p tsconfig.json && run-benchmark 2 3"
   },
   "engines": {
     "node": ">=20"
   },
-  "engineStrict": true
+  "engineStrict": true,
+  "devDependencies": {
+    "ts-node": "^10.9.2",
```
Comment (Idrinth) [inline examples/reqres.in/package.json:18]: tsc may be easier

File: examples/reqres.in/src/routes/main/create-user.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-module.exports = (apiRootUrl,) => ({
+export default (apiRootUrl: string,) => ({
```
[the remark this answers is not in the document]
Comment (Idrinth) [reply examples/reqres.in/src/routes/main/create-user.ts:1]: @coderabbitai the trailing comma is a requirement in this organisation.

**Review by Idrinth — CHANGES_REQUESTED** [review]
looks fine for the most part, please check the actions.

**Comment by Idrinth** [comment]
> @Idrinth I did not understand the check about the "examples target", how can I "inform it"?

already done for existing ones

**Comment by Idrinth** [comment]
please move it over to the example repo :)

**Pull request closed without merging**